### PR TITLE
Fix issue95

### PR DESCRIPTION
**Pull request by dnkolegov**
This PR addresses issues https://github.com/consensus-shipyard/lotus/issues/98 and https://github.com/consensus-shipyard/lotus/issues/95. [Tests](https://app.circleci.com/pipelines/github/consensus-shipyard/lotus?branch=issue95) have succeeded.

### CODE REVIEW

**Review by matejpavlovic — APPROVED** [review]
LGTM.
Don't quite understand how the whole business with Byzantine twins works, but that doesn't seem to be the point here.
Thanks for the fix!

**Comment by dnkolegov** [comment]
> LGTM. Don't quite understand how the whole business with Byzantine twins works, but that doesn't seem to be the point here. Thanks for the fix!

I did not understand why you mentioned "twins", but that was my first attempt to add a very simple test from this [paper](https://arxiv.org/pdf/2004.10617.pdf).

**Comment by matejpavlovic** [comment]
Added the paper to my reading list, will have a look at it later. For now, I think we can merge this, unless @adlrocha still wants to review it.

**Comment by dnkolegov** [comment]
> Added the paper to my reading list, will have a look at it later. For now, I think we can merge this, unless @adlrocha still wants to review it.

I believe @adlrocha must review this PR.

File: chain/consensus/mir/state_manager.go
Comment (adlrocha) [inline chain/consensus/mir/state_manager.go:289]: While this makes sense I don't understand why it makes a difference. Do you mind explaining me why this fixes the issue?

**Review by adlrocha — APPROVED** [review]
LGTM, but as mentioned in the comments I would love to understand why this works before merging (I see that by using the expected height instead of the heaviest one we remove a source of potential races, but is this the case?). Thanks.

File: chain/consensus/mir/state_manager.go
Comment (adlrocha) [reply chain/consensus/mir/state_manager.go:289]: Oh, gotcha!!! Nice fix then, folks. This was a hard one.

Comment (dnkolegov) [reply chain/consensus/mir/state_manager.go:289]: Thank you, @matejpavlovic !

**Pull request merged**
